### PR TITLE
[SCB-868] Add Kamon metrics to Alpha Server

### PR DESCRIPTION
**Pull request by maheshrajus**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: alpha/pom.xml [hunk 1]
```diff
@@ -35,6 +35,27 @@
...
+    <dependency>
+      <groupId>io.kamon</groupId>
+      <artifactId>kamon-core_2.12</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.kamon</groupId>
+      <artifactId>kamon-annotation_2.12</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.servicecomb.saga</groupId>
```
Comment (WillemJiang) [inline alpha/pom.xml]: alpha doesn't use saga-core , we should not add this dependency.

Comment (maheshrajus) [reply alpha/pom.xml]: ok.done

File: alpha/pom.xml [hunk 2]
```diff
@@ -35,24 +35,21 @@
     <module>alpha-server</module>
   </modules>
 
+  <properties>
+    <kamon.version>0.6.0</kamon.version>
```
Comment (WillemJiang) [inline alpha/pom.xml]: We normally put the dependency properties into parent pom, so the other module could share the same properties.

File: alpha/pom.xml [hunk 3]
```diff
@@ -35,24 +35,21 @@
...
+    <akka.version>2.3.12</akka.version>
+  </properties>
+
   <dependencies>
-    <dependency>
+   <dependency>
       <groupId>io.kamon</groupId>
-      <artifactId>kamon-core_2.12</artifactId>
+      <artifactId>kamon-core_2.11</artifactId>
```
Comment (WillemJiang) [inline alpha/pom.xml]: Why do you change the scala version from 2.12 to 2.11?

File: alpha/alpha-server/src/main/resources/application.conf [hunk 4]
```diff
@@ -24,31 +24,37 @@ kamon {
...
+      akka-dispatcher.excludes = []
     }
   }
 
   show-aspectj-missing-warning = no
 
   modules {
     kamon-annotation {
-      auto-start = no
+      auto-start = yes
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/resources/application.conf]: Why did you change these settings?

File: alpha/pom.xml [hunk 5]
```diff
@@ -75,20 +72,42 @@
...
-          <artifactId>kamon-autoweave_2.12</artifactId>
-          <version>0.6.5</version>
+          <artifactId>kamon-statsd_2.11</artifactId>
+          <version>${kamon.version}</version>
         </dependency>
         <dependency>
-          <groupId>org.aspectj</groupId>
-          <artifactId>aspectjweaver</artifactId>
+          <groupId>com.typesafe.akka</groupId>
+          <artifactId>akka-actor_2.11</artifactId>
```
Comment (WillemJiang) [inline alpha/pom.xml]: Current Alpha Server doesn't use Akka, why did you put these dependencies.

File: alpha/alpha-server/src/main/resources/application.conf (hunk 4 above)
Comment (maheshrajus) [reply alpha/alpha-server/src/main/resources/application.conf]: while running the alpha server I have to provide auto-start yes for kamon.  I put here yes and forgot to revert. i will make by default "no" and when user required then they will pass argument "yes".

File: alpha/pom.xml (hunk 5 above)
Comment (maheshrajus) [reply alpha/pom.xml]: saga-core-akka support is there right so i thought akka support is there. i will remove unwated dependency.

File: alpha/pom.xml (hunk 3 above)
Comment (maheshrajus) [reply alpha/pom.xml]: kamon-autoweave  dependency 0.6.0 not available for 2.12 version. So i made all kamon dependency to 2.11 version so that all kamon classes should be in single version.

Comment (maheshrajus) [reply alpha/pom.xml]: I will modify to latest version which is 2.12 only

File: alpha/pom.xml [hunk 6]
```diff
@@ -35,6 +35,21 @@
     <module>alpha-server</module>
   </modules>
 
+  <dependencies>
```
Comment (WillemJiang) [inline alpha/pom.xml]: We should put these dependencies into alpha-server.

**Comment by WillemJiang** [comment]
Merged the patch by apply some changes on pom files with thanks to maheshrajus.
